demo/futures/test_futures.py

```diff
@@ -1305,6 +1305,9 @@ def transform(a):
     if version == (4,0,2) and sys.platform=='darwin':
         SKIP_POOL_TEST = True
 if name == 'MPICH':
+    if sys.platform == 'darwin':
+        if version >= (3, 4) and version < (4, 0):
+            SKIP_POOL_TEST = True
```

Comment:
Is this due to https://github.com/pmodels/mpich/issues/5149?

Reply:
Yes. Current MPICH 3.4.1 from Homebrew on macOS Catalina does not work with the `mpi4py.futures` tests that involve `MPI_Comm_spawn()`. I have no idea what's going on, I cannot reproduce the problem in Linux. So I just disabled the test hoping for future resolution in MPICH 4.0.